idseq_pipeline/commands/accessionid2seq_functions.py

```diff
@@ -42,6 +43,59 @@ def parse_reads(annotated_fasta, db_type):
                             read2seq[ma.group(4).rstrip()] = [sequence.rstrip(), ma.group(1), ma.group(2), ma.group(3)]
     return read2seq
 
+def compress_coverage(coverage):
+    keys = sorted(coverage.keys())
+    if len(keys) <= 1:
+        return coverage
+    output = {}
+
+    start = keys[0]
+    current = start
+    val = coverage[start]
+
+    for k in keys[1:]:
+        if (k-current) == 1 and coverage[k] == val:
+            current = k
+        else:
+            output["%d-%d" % (start, current)] = val
+            start = k
+            current = k
+            val = coverage[k]
+
+    output["%d-%d" % (start, current)] = val
+    return output
+
+
+def calculate_alignment_coverage(alignment_data):
+    ref_len = alignment_data['ref_seq_len']
+    coverage = defaultdict(lambda: 0) # implement it in a dumb way for now. change later
+    output = {
+        'ref_seq_len': ref_len,
+        'total_read_length': 0,
+        'total_aligned_length' : 0,
+        'total_mismatched_length' : 0,
+        'num_reads': 0
+    }
+    if ref_len == 0:
+        return output
+    reads = alignment_data['reads']
+    for read in reads:
+        seq = read[1]
+        m8_metrics = read[2]
+        ref_start = int(m8_metrics[-4])
+        ref_end = int(m8_metrics[-3])
+        if ref_start > ref_end: # SWAP
+            (ref_start, ref_end) = (ref_end, ref_start)
+        ref_start -= 1
+        output['total_read_length'] += len(seq)
+        output['total_aligned_length'] += (ref_end - ref_start)
+        output['total_mismatched_length'] += int(m8_metrics[2])
+        output['num_reads'] += 1
+        for bp in range(ref_start, ref_end):
```

Comment:
Either this range is incorrect or the total_aligned_length computation above is incorrect.